UnitTests/StreamCopierTest.cs

```diff
@@ -140,92 +146,94 @@ public void CopyStream_UserCancels_ReturnsFalse ()
-      inputStub.Stub (_ => _.Read (null, 0, 0)).IgnoreArguments().Return (0);
+      var inputStub = new Mock<Stream>();
+      inputStub.SetupGet (_ => _.CanSeek).Returns (true);
+      inputStub.SetupGet (_ => _.Length).Returns (11);
+      var sequence = new MockSequence();
+      inputStub.InSequence (sequence).Setup (_ => _.Read (It.IsAny<Byte[]>(), It.IsAny<int>(), It.IsAny<int>())).Returns (13);
+      inputStub.InSequence (sequence).Setup (_ => _.Read (It.IsAny<Byte[]>(), It.IsAny<int>(), It.IsAny<int>())).Returns (0);
 
       Assert.That (
-          () => _streamCopier.CopyStream (inputStub, new MemoryStream()),
+          () => _streamCopier.CopyStream (inputStub.Object, new MemoryStream()),
           Throws.InstanceOf<IOException>().With.Message.EqualTo ("The stream returned more data than the stream length (11 bytes) specified."));
     }
 
     [Test]
     public void CopyStream_StreamReturnsMoreDataThanMaximumLengthAndStreamLength_ThrowsIOException ()
     {
-      Stream inputStub = MockRepository.GenerateStub<Stream>();
-      inputStub.Stub (_ => _.CanSeek).Return (true);
-      inputStub.Stub (_ => _.Length).Return (11);
-      inputStub.Expect (_ => _.Read (null, 0, 0)).IgnoreArguments().Return (13);
+      var inputStub = new Mock<Stream>();
+      inputStub.SetupGet (_ => _.CanSeek).Returns (true);
+      inputStub.SetupGet (_ => _.Length).Returns (11);
+      inputStub.Setup (_ => _.Read (It.IsAny<Byte[]>(), It.IsAny<int>(), It.IsAny<int>())).Returns (13).Verifiable();
 
       Assert.That (
-          () => _streamCopier.CopyStream (inputStub, new MemoryStream(), 12),
+          () => _streamCopier.CopyStream (inputStub.Object, new MemoryStream(), 12),
           Throws.InstanceOf<IOException>().With.Message.EqualTo ("The stream returned more data (13 bytes) than the specified maximum (12 bytes)."));
     }
 
-    private void SetupReadExpectation (Stream streamMock, IEnumerable<byte> bytes)
+    private void SetupReadExpectation (Mock<Stream> streamMock, MockSequence mockSequence, IEnumerable<byte> bytes)
     {
       var byteArray = bytes.ToArray();
 
       streamMock
-          .Expect (s => s.Read (null, 0, 0))
-          .IgnoreArguments()
-          .WhenCalled (invocation =>
-          {
-            var buffer = (byte[]) invocation.Arguments[0];
-            var offset = (int) invocation.Arguments[1];
-
-            for (int i = 0; i < byteArray.Length; ++i)
-              buffer[offset + i] = byteArray[i];
-          })
-          .Return (byteArray.Length)
-          .Repeat.Once();
+         .InSequence (mockSequence)
+         .Setup (s => s.Read (It.IsAny<byte[]>(), It.IsAny<int>(), It.IsAny<int>()))
+         .Callback (
+              (byte[] bufferA, int offsetA, int _) =>
+              {
+                var buffer = bufferA; //REVIEW Parameters are no longer passed by an Array -> remove variables and directly access parameters?
+                var offset = offsetA;
```

Comment:
Yes, use the parameters directly. It doesn't complicate the review since it's a multiline hunk anyway. Also names like `bufferA` aren't that great